### DISCUSSION TITLE
NER on dates

### DISCUSSION THREAD

**Discussion by Phat-Loc** [explosion/spaCy#8082] [Help: Best practices]
What is the best practice for NER of dates? Will the NER pipeline component be able to distinguish different date entities just by context or should I include the context in the span of an entity and do post processing? For example

There was a meeting on March 31, 2021.

If I wanted an NER of meeting date separate from other dates should I use the span of "meeting on March 31, 2021" or just "March 31, 2021"? Will the NER pipeline component pick up from examples the context of date to distinguish the different NER dates or should I explicitly put the context in the entity? Thanks!

**Comment by polm** [comment]
Don't include the context, that makes it harder to learn. There are some kinds of NER models that handle that better, but spaCy's works best for tokens with consistent boundaries, as mentioned [in the docs](https://spacy.io/api/entityrecognizer). 

> If I wanted an NER of meeting date separate from other dates

This is a very hard problem to model as NER. Generally with this problem it's easier to use a high-quality existing model and refine the results using the dependency parse or other tools. For example, if you just treat all DATE entities in sentences with the word "meeting" as MEETING_DATE, would that handle your use case? If it's not perfect how close is it?

[This talk](https://github.com/explosion/talks/blob/master/2018-10-31_Building-New-NLP-Solutions-with-spaCy-and-Prodigy.pdf.pdf) might be a useful reference.
